quixstreams/state/metadata.py

```diff
@@ -1,13 +1,14 @@
-from typing import NewType
+import enum
 
 PREFIX_SEPARATOR = b"|"
 
 CHANGELOG_CF_MESSAGE_HEADER = "__column_family__"
 CHANGELOG_PROCESSED_OFFSET_MESSAGE_HEADER = "__processed_tp_offset__"
 METADATA_CF_NAME = "__metadata__"
 
-Undefined = NewType("Undefined", object)
-UNDEFINED = Undefined(object())
-DELETED = Undefined(object())
-
 DEFAULT_PREFIX = b""
+
+
+class markers(enum.Enum):
```

Comment:
That must not be lowercase plural but uppercase singular. Type hint like `Union[bytes, markers]` must be `Union[bytes, Marker]`.

```suggestion
class Marker(enum.Enum):
```